includes/class-gp-translation-helpers.php

```diff
@@ -345,22 +345,53 @@ public function register_reject_feedback_js( $template, $translation_set ) {
 	public function reject_with_feedback() {
 		check_ajax_referer( 'gp_reject_feedback', 'nonce' );
 
-		$helper_discussion = new Helper_Translation_Discussion();
-		$locale_slug       = $helper_discussion->sanitize_comment_locale( sanitize_text_field( $_POST['data']['locale_slug'] ) );
-		$translation_id    = $helper_discussion->sanitize_translation_id( intval( $_POST['data']['translation_id'] ) );
-		$original_id       = $_POST['data']['original_id'];
-		$reject_reason     = ! empty( $_POST['data']['reason'] ) ? $_POST['data']['reason'] : array( 'other' );
-		$reject_reason     = array_map( 'sanitize_text_field', $reject_reason );
-		$reject_comment    = sanitize_text_field( $_POST['data']['comment'] );
+		$helper_discussion    = new Helper_Translation_Discussion();
+		$locale_slug          = $helper_discussion->sanitize_comment_locale( sanitize_text_field( $_POST['data']['locale_slug'] ) );
+		$translation_id_array = array_map( array( $helper_discussion, 'sanitize_translation_id' ), $_POST['data']['translation_id'] );
+		$original_id_array    = array_map( array( $helper_discussion, 'sanitize_original_id' ), $_POST['data']['original_id'] );
+		$reject_reason        = ! empty( $_POST['data']['reason'] ) ? $_POST['data']['reason'] : array( 'other' );
+		$reject_reason        = array_map( 'sanitize_text_field', $reject_reason );
+		$reject_comment       = sanitize_text_field( $_POST['data']['comment'] );
+
+		if ( ! $locale_slug || ! $translation_id_array || ! $original_id_array || ( ! $reject_reason && ! $reject_comment ) ) {
```

Comment:
I think we should allow the rejection of strings without any reason or comment, as is currently done, so I would change these two conditions
```suggestion
		if ( ! $locale_slug || ! $translation_id_array || ! $original_id_array ) {
```

Reply:
The rejection itself is actioned in the reject-feedback.js, here we are checking if there's no reason and no comment, hence nothing should be done.

Reply:
I think the system should allow to reject strings without reason and without comment, as it was being done until now.

![image](https://user-images.githubusercontent.com/1667814/157285862-00bb05f0-d9d9-446c-8840-6df606d73716.png)

Reply:
Thanks for catching this, fixed!